### ISSUE TITLE
Change py_config scope to `function`

### ISSUE THREAD

**Issue by ElSnoMan** [ElSnoMan/pyleniumio#147]
## Problem

There isn't a record as to _why_ this was changed to a `session` scope. Ideally, this should have the `function` scope so each test has its own config.

## Potential Solution(s)

Change the scope to `function`

```python
@pytest.fixture(scope='function')
def py_config(project_root, request) -> PyleniumConfig:
    """ Initialize a PyleniumConfig for each test

    1. This starts by deserializing the user-created pylenium.json from the Project Root.
    2. If that file is not found, then proceed with Pylenium Defaults.
    3. Then any CLI arguments override their respective key/values.
    """
```

**Comment by rnestler** [comment]
Isn't it bad for performance if every test needs to re-read the configuration file from disk?

**Comment by rnestler** [comment]
Maybe we could parse it only once and return a `deepcopy` for earch test function?

**Comment by ElSnoMan** [comment]
> Maybe we could parse it only once and return a `deepcopy` for each test function?

I really like that idea! The end goal is being able to globally configure all tests easily (via CLI or pylenium.json) while allowing a single test to change a config just for its scope. 

For example:

```bash
$ pytest --browser=chrome
```

☝🏽 This will have all tests run against a Chrome browser, but I may have a single test that uses selenium-wire to capture and test underlying requests made by the browser. Setting selenium-wire values in pylenium.json would affect all tests, so we'd want to avoid that with something like:

```python
def test_website_requests(py):
    # Set selenium-wire options just for this test
    options = {
        'proxy': {
            'http': 'http://192.168.10.100:8888/',
            'https': 'https://192.168.10.100:8888/',
            'no_proxy': 'localhost,127.0.0.1'
        }
    }
    py.config.driver.seleniumwire_options = options
    py.visit(URL)
    # do stuff and test stuff
```

I haven't tested if this works with the fixture being `session` scoped, but I'm open to ideas and pull requests! Do you want to give it a try?

**Comment by rnestler** [comment]
> Do you want to give it a try?

Already did. See https://github.com/ElSnoMan/pyleniumio/pull/247 :slightly_smiling_face:

**Comment by ElSnoMan** [comment]
Resolved in `pyleniumio, v1.15.2+`
